File: PublicServiceRequest/Models/RequestTaxpayer.cs
namespace CustomerService_PSR.Models
{
    using System;
    using System.Collections.Generic;
    using System.ComponentModel.DataAnnotations;
    using System.ComponentModel.DataAnnotations.Schema;
    using System.Data.Entity.Spatial;

    [Table("PSR.RequestTaxpayer")]
    public partial class RequestTaxpayer : BaseClass
    {
        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Usage", "CA2214:DoNotCallOverridableMethodsInConstructors")]
        public RequestTaxpayer()
        {
            Requests = new HashSet<Request>();
        }

        public int RequestTaxpayerID { get; set; }

        public int? RequestTaxpayerPreferredLanguageID { get; set; }

        [StringLength(50)]
        public string FirstName { get; set; }

        [StringLength(50)]
        public string LastName { get; set; }

        [StringLength(1)]
        public string MiddleInitial { get; set; }

        [DatabaseGenerated(DatabaseGeneratedOption.Computed)]
        [StringLength(103)]
        public string FormattedTaxpayerName { get; set; }

        [StringLength(30)]
        public string PhoneNo { get; set; }

        [StringLength(30)]
        public string PhoneNoType { get; set; }

        [StringLength(30)]
        public string AltPhoneNo1 { get; set; }

        [StringLength(30)]
        public string AltPhoneNo1Type { get; set; }

        [StringLength(30)]
        public string AltPhoneNo2 { get; set; }

        [StringLength(30)]
        public string AltPhoneNo2Type { get; set; }

        [DatabaseGenerated(DatabaseGeneratedOption.Computed)]
        [StringLength(14)]
        public string FormattedPhoneNo { get; set; }

        [DatabaseGenerated(DatabaseGeneratedOption.Computed)]
        [StringLength(14)]
        public string FormattedAltPhoneNo1 { get; set; }

        [DatabaseGenerated(DatabaseGeneratedOption.Computed)]
        [StringLength(14)]
        public string FormattedAltPhoneNo2 { get; set; }

        [StringLength(100)]
        public string Email { get; set; }

        [StringLength(100)]
        public string AltEmail { get; set; }

        public bool? SystemReservedFlag { get; set; }

        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Usage", "CA2227:CollectionPropertiesShouldBeReadOnly")]
        public virtual ICollection<Request> Requests { get; set; }

        public virtual RequestTaxpayerPreferredLanguage RequestTaxpayerPreferredLanguage { get; set; }
    }
}

File: PublicServiceRequest/app/domain/requestTaxpayer.ts
module app.domain {

    export interface IRequestTaxpayer {
        RequestTaxpayerID: number;
        RequestTaxpayerPreferredLanguageID: number;
        FirstName: string;
        MiddleInitial: string;
        LastName: string;
        FormattedTaxpayerName: string;
        PhoneNo: string;
        FormattedPhoneNo: string;
        PhoneNoType: string;
        AltPhoneNo1: string;
        AltPhoneNo1Type: string;
        AltPhoneNo2: string;
        AltPhoneNo2Type: string;
        Email: string;
        AltEmail: string,
        SystemReservedFlag: boolean,
        AdmIsActive: string;
        AdmUserAdded: number;
        AdmUserAddedFullName: string;
        AdmDateAdded: string;
        AdmUserModified: number;
        AdmUserModifiedFullName: string;
        AdmDateModified: string;
        RemainAnonymousFlag: boolean;
    }

    export class RequestTaxpayer implements IRequestTaxpayer {
        constructor(public RequestTaxpayerID: number,
            public RequestTaxpayerPreferredLanguageID: number,
            public FirstName: string,
            public MiddleInitial: string,
            public LastName: string,
            public FormattedTaxpayerName: string,
            public PhoneNo: string,
            public FormattedPhoneNo: string,
            public PhoneNoType: string,
            public AltPhoneNo1: string,
            public AltPhoneNo1Type: string,
            public AltPhoneNo2: string,
            public AltPhoneNo2Type: string,
            public Email: string,
            public AltEmail: string,
            public SystemReservedFlag: boolean,
            public AdmIsActive: string,
            public AdmUserAdded: number,
            public AdmUserAddedFullName: string,
            public AdmDateAdded: string,
            public AdmUserModified: number,
            public AdmUserModifiedFullName: string,
            public AdmDateModified: string,
            public RemainAnonymousFlag: boolean) { }
    }
}